docs/rest-api.md

```diff
@@ -0,0 +1,117 @@
+
+## API
+
+The following table shows all versions of REST API we have for now. You can also find the swagger file for each version of REST API.
+
+API Version | Status | Port | Spec
+---|---|---|---
+`1.0-preview` | Available | 5002 | [swagger](./swagger/v1-preview.json)
+`1.0` | *Not Available until GA* | Default | [swagger](./swagger/v1.json)
+
+Available APIs of each version are listed as following.
+
+API | `1.0-preview` | `1.0`
+---|---|---
+[Broadcast to all](#broadcast) | :heavy_check_mark: | :heavy_check_mark:
+[Broadcast to a group](#broadcast-group) | :heavy_check_mark: | :heavy_check_mark:
+Broadcast to a few groups | :heavy_check_mark: (Deprecated) | `N/A`
+[Send to a user](#send-user) | :heavy_check_mark: | :heavy_check_mark:
+Send to a few users | :heavy_check_mark: (Deprecated) | `N/A`
+[Add a user to a group](#add-user-to-group) | `N/A` | :heavy_check_mark:
+[Remove a user from a group](#remove-user-from-group) | `N/A` | :heavy_check_mark:
+
+<a name="broadcast"></a>
+### Broadcast message to all clients
+
+API Version | HTTP Method | Request URL | Request Body
+---|---|---|---
+`1.0-preview` | `POST` | `https://<instance-name>.service.signalr.net:5002/api/v1-preview/hub/<hub-name>` | `{ "target":"<method-name>", "arguments":[ ... ] }`
+`1.0` | `POST` | `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>` | Same as above
+
+<a name="broadcast-group"></a>
+### Broadcast message to a group
+
+API Version | HTTP Method | Request URL | Request Body
+---|---|---|---
+`1.0-preview` | `POST` | `https://<instance-name>.service.signalr.net:5002/api/v1-preview/hub/<hub-name>/group/<group-name>` | `{ "target":"<method-name>", "arguments":[ ... ] }`
+`1.0` | `POST` | `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>/groups/<group-name>` | Same as above
+
+<a name="send-user"></a>
+### Send message to a user
+
+API Version | HTTP Method | Request URL | Request Body
+---|---|---|---
+`1.0-preview` | `POST` | `https://<instance-name>.service.signalr.net:5002/api/v1-preview/hub/<hub-name>/user/<user-id>` | `{ "target":"<method-name>", "arguments":[ ... ] }`
+`1.0` | `POST` | `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>/users/<user-id>` | Same as above
+
+<a name="add-user-to-group"></a>
+### Add a user to a group
+
+API Version | HTTP Method | Request URL
+---|---|---
+`1.0` | `PUT` | `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>/groups/<group-name>/users/<user-id>`
+
+<a name="remove-user-from-group"></a>
+### Remove a user from a group
+
+API Version | HTTP Method | Request URL
+---|---|---
+`1.0` | `DELETE` | `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>/groups/<group-name>/users/<user-id>`
+
+## Using REST API
+
+<a name="user-api"></a>
+### User-related REST API
+
+In order to call user-related REST API, each of your clients should identify themselves to Azure SignalR Service.

```

Comment:
Add more details about how this can be done (through negotiate). Add an example about negotiate response.

Reply:
Negotiate will be explained in details in my next PR. Will reference it then,